### ISSUE TITLE
[REVIEW]: Developing and deploying an integrated workshop curriculum teaching computational skills for reproducible research

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#144]
**Submitting author:** @pschloss (<a href="http://orcid.org/0000-0002-6935-4275">Patrick D Schloss</a>)
**Repository:** <a href="https://github.com/UMCarpentries/intro-curriculum-r/" target ="_blank">https://github.com/UMCarpentries/intro-curriculum-r/</a>
**Version:** v1.0.0
**Editor:** @magsol
**Reviewer:** @kdillmcfarland, @tracykteal
**Archive:** <a href="https://doi.org/10.5281/zenodo.5812479" target="_blank">10.5281/zenodo.5812479</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://jose.theoj.org/papers/4649d22db51c8150274f83f806792dc9/status.svg)](https://jose.theoj.org/papers/4649d22db51c8150274f83f806792dc9)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/4649d22db51c8150274f83f806792dc9"><img src="https://jose.theoj.org/papers/4649d22db51c8150274f83f806792dc9/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/4649d22db51c8150274f83f806792dc9/status.svg)](https://jose.theoj.org/papers/4649d22db51c8150274f83f806792dc9)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/UMCarpentries/intro-curriculum-r/tree/gh-pages/paper" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@kdillmcfarland & @tracykteal, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @magsol know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/UMCarpentries/intro-curriculum-r/tree/gh-pages/paper">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v1.0.0
- [x] **Authorship:** Has the submitting author (@pschloss) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/UMCarpentries/intro-curriculum-r/tree/gh-pages/paper">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v1.0.0
- [x] **Authorship:** Has the submitting author (@pschloss) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by kdillmcfarland** [comment]
Documentation

- Request for improved Linux installation https://github.com/UMCarpentries/intro-curriculum-r/issues/147
- Request for expanded statement of purpose https://github.com/UMCarpentries/intro-curriculum-r/issues/148

**Comment by magsol** [comment]
Thanks Kimberly! @pschloss have you had a chance to review the feedback from Kimberly? It looks like this is about it.

@tracykteal how is your review going? Any chance of wrapping this up before the holidays?

**Comment by zenalapp** [comment]
We have reviewed and implemented all changes suggested by @kdillmcfarland.

**Comment by magsol** [comment]
@tracykteal Hi Tracy! We're waiting on your review before we can move forward with this submission. Any chance of completing your review soon?

**Comment by magsol** [comment]
Hi @tracykteal! Haven't heard from you in several weeks now. Are you able to complete your review soon, or should we look for another reviewer? Please advise as soon as you can.

**Comment by tracykteal** [comment]
Thanks! I've gone through my review, and I'm very sorry for the delay. I also recommend this paper for publication without revision. This is an excellent lesson that creates a clear 2 day curriculum that is easy to use for instructors and learners, reducing the overhead of getting started. 

This is a lesson based on Software Carpentry and developed by the UM Carpentries instance, so it makes sense to refer to it as 'Carpentries'. It's not officially Carpentries though, given that it's not in a Carpentries space, so some of the references to Carpentries infrastructure aren't relevant. I put in one PR to address.  

- [ ] https://github.com/UMCarpentries/intro-curriculum-r/pull/169

I also created an issue with some potential clarification language on the index page.  

- [ ] https://github.com/UMCarpentries/intro-curriculum-r/issues/170

The authors may have already had some conversations with Carpentries about naming and acknowledgement though, so if so, please just use whatever they've recommended!

**Comment by zenalapp** [comment]
Thanks, @tracykteal! We have merged your PR and updated the language about our connection to Carpentries lessons.

@magsol is there anything else we need to do?

**Comment by magsol** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #144 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by magsol** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00144/jose.00144/10.21105.jose.00144.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00144/jose.00144/10.21105.jose.00144.pdf) :page_facing_up: :point_left:

**Comment by magsol** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.01686 is OK
- 10.12688/f1000research.3-62.v2 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by magsol** [comment]
@zenalapp This looks great! At this point I'd ask for you to create a *tagged release* (only if you don't already have one, or have made updates since your last one) and *archive*, and post both the release **version number** and the **archive DOI** here, and I can take it from there.

**Comment by kelly-sovacool** [comment]
@magsol I just released version `1.0.0` with DOI `10.5281/zenodo.5812479`.

**Comment by magsol** [comment]
@whedon set 10.5281/zenodo.5812479 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5812479" target="_blank">10.5281/zenodo.5812479</a> is the archive.

**Comment by magsol** [comment]
@whedon set v1.0.0 as version

**Comment by whedon** [comment]
OK. v1.0.0 is the version.

**Comment by magsol** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.01686 is OK
- 10.12688/f1000research.3-62.v2 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/jose-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/74

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/74, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@pschloss – Could you try to fix the horizontal badness on URLs that appear in the References? Maybe using a `\url` inside the `howpublished` field of the `.bib` entry?

<img width="932" alt="Screen Shot 2022-01-01 at 7 10 33 PM" src="https://user-images.githubusercontent.com/4800109/147861164-46c44281-6ff9-4b0d-8d66-3ffb34ad2ecb.png">

**Comment by zenalapp** [comment]
Just merged the PR from @kelly-sovacool to fix this.

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00144/jose.00144/10.21105.jose.00144.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00144/jose.00144/10.21105.jose.00144.pdf) :page_facing_up: :point_left:

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/77
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jose.00144](https://doi.org/10.21105/jose.00144)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @zenalapp and co-authors, your article is published in JOSE! 🚀 

Huge thanks to our Editor: @magsol and the Reviewers: @kdillmcfarland, @tracykteal — we appreciate your contribution to this adventure in publishing 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00144/status.svg)](https://doi.org/10.21105/jose.00144)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00144">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00144/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00144/status.svg
   :target: https://doi.org/10.21105/jose.00144
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00144/status.svg)](https://doi.org/10.21105/jose.00144)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-jose

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
